### DISCUSSION TITLE
Platform Cache Behavior

### DISCUSSION THREAD

**Discussion by ghost** [jamessimone/salesforce-round-robin#12] [General]
Hey @jamessimone 

Just troubleshooted a Platform cache issue that I accidentally created for myself using your Round Robin (great app so far, cheers!).
I was reading your wiki about using platform cache - and how a platform cache entry is required and created as part of this package - and when I went and had a look i noticed 2 things:

1. My org didn't have any allocations assigned (they were all 0's) for Round Robin
2. When i clicked 'edit' i wasn't able to adjust the 'Provider Free' allocation - only Org cache under session cache/org cache.

so i thought i had to allocate some capacity - so i added 1mb in both Session and Org.

Then the round Robin assignment broke....
so i removed my newly allocated mb's - and it started working again.

Are you able to explain how that works? I'm probably doing it wrong, but just in case i have identified a bug.

cheers

**Comment by ghost** [comment]
Sorry - when i say "the round Robin assignment broke" - it did the following:

1. It no longer updated the Owner - it just assigned me as the owner (as the record creator i assume).
2. the 'Index' field on the Round Robin Custom Setting didn't change/update
3. BUT the 'Last Updated' field on the Round Robin Custom Setting did keep updating every time i created a new record that was supposed to be assigned.

**Reply by jamessimone** [reply]
I'll have to have a look, thank you for bringing this to my attention!

**Reply by jamessimone** [reply]
@jmsctm OK, this behavior is explainable by the issue you reported in #13. It will be fixed when that update goes out, shortly.

**Reply by jamessimone** [reply]
On the subject of cache - if installing the package _does not_ automatically assign org cache, you'll want to allocate at least 1 mb to it under the `Organization` header. Adding session cache won't do anything, as Round Robin doesn't use Session Cache. Hope that helps!

**Reply by ghost** [reply]
Thank you for the advice @jamessimone - I'll allocate 1mb under Organization 👍

**Reply by jamessimone** [reply]
Sounds good!

**Reply by ghost** [reply]
Hey @jamessimone 

I've installed the new package into my Dev environment - and the cache still appears to cause problems when i allocate 1mb. I've attached a screenshot of my allocation.

after upgrading, i tested the behaviour with Zero allocation - and the round robin was working as expected. Then i added 1mb to Org Cache and the very next allocation worked as expected (grabbed the next user), but then it got stuck on that user. If i clear the platform cache using the 'clear cache' button in the screenshot, it will again grab the next user, but then get stuck on that user 
<img width="1097" alt="Round Robin Cache" src="https://github.com/jamessimone/salesforce-round-robin/assets/106937074/7a2865c1-9e45-45fd-be18-20e2cf81824b">

**Reply by ghost** [reply]
it works correctly without any cache allocated - screenshot of working cache below

<img width="1098" alt="Working Cache Round Robin" src="https://github.com/jamessimone/salesforce-round-robin/assets/106937074/f051cdf2-491d-4e3c-be4a-aaf1168aefd9">

**Reply by jamessimone** [reply]
Ok, I will take a look! Thanks for the details, they always help.

**Reply by jamessimone** [reply]
@jmsctm I haven't forgotten about you - it's been a busy few weeks, is all. I was able to spend a few hours investigating the issue today, and I've created #15 as a result.

**Reply by ghost** [reply]
thanks for the follow up @jamessimone - hope you don't feel any pressure. Appreciate the fact that you provide this sort of product/project at all outside of regular working life!

**Reply by jamessimone** [reply]
Of course. I'm happy to help! The latest version should fix this up nicely for you - definitely don't hesitate to let me know if there's anything else amiss.
